### ISSUE TITLE
Make Test Stand Client into .exe 

### ISSUE THREAD

**Issue by CEngelbrecht** [SDSURocketProject/Test-Stand-DAQ#5]
Use py2exe to make the client an executable.

**Comment by colanzio5** [comment]
Not sure if we can do this with python3. Py2exe only seems to be supported by python 2.7 and below.

**Comment by alefir** [comment]
Why would we want this?

**Comment by colanzio5** [comment]
Its waaaaay easier to click a .exe than potentially have to install every single dependency.

**Comment by alefir** [comment]
It would also take away the cross-platform benefit of python, and make it a lot harder to make on-the-fly changes by forcing compiled releases.

**Comment by colanzio5** [comment]
Not really, you just run it as a python script if you want to run it on Linux on macos. Otherwise we can distribute it as a .exe file for windows users.

**Comment by colanzio5** [comment]
**Linux or macos**

**Comment by alefir** [comment]
Windows users can also run it as the original python, which leaves me wondering what the benefit of compiling it into an exe would be

**Comment by colanzio5** [comment]
Not having to installing the dependencies if we're just trying to put a and play the client, like I said earlier. During launch, we really don't want to have to mess around with stuff like that if we're on a short launch window. Usually if we're posting an issue to the git, me or Christian have a good reason.

**Comment by CEngelbrecht** [comment]
It's currently a priority because we're using a data acquisition program (WinDAQ) that's windows only, and we most definitely shouldn't be making on the fly changes. However, during yesterday's cryo test, I had to do just that (adding more buttons to the GUI). So if someone outside of Avionics wanted to use the client as is, there should be .exe on the AvTop desktop, but there should also be the python code available for on the fly builds if necessary.

**Comment by CEngelbrecht** [comment]
It's currently a priority because we're using a data acquisition program (WinDAQ) that's windows only, and we most definitely shouldn't be making on the fly changes. However, during yesterday's cryo test, I had to do just that (adding more buttons to the GUI). So if someone outside of Avionics wanted to use the client as is, there should be .exe on the AvTop desktop, but there should also be the python code available for on the fly builds if necessary.

**Comment by CEngelbrecht** [comment]
It's currently a priority because we're using a data acquisition program (WinDAQ) that's windows only, and we most definitely shouldn't be making on the fly changes. However, during yesterday's cryo test, I had to do just that (adding more buttons to the GUI). So if someone outside of Avionics wanted to use the client as is, there should be .exe on the AvTop desktop, but there should also be the python code available for on the fly builds if necessary.

**Comment by CEngelbrecht** [comment]
Also, if the AvTop is somehow lost/damaged/unuseable at launch or static fire days, we can download the .exe from here and use on anyone's computer.

**Comment by colanzio5** [comment]
For portability, I'd like to take the opportunity to do a JavaScript client. I can keep looking into making the python script an exe, but you can build electron (html/css/just) packages into cross platform apps super easily, so it kinda makes sense. I'll just start a new branch and let you know if/when I get it in a good spot functionally.

**Comment by waltoncade** [comment]
I think the exe is a good idea, I will work on getting it done. @alefir When we next meet, I am going to need more help navigating git and contributing to making changes to the master branch. @colanzio5 I want to do a JavaScript client as well. I think that we should do so for the flight control of the Far Mars rocket, and not the test stand. I think that it will be most useful in data visualization, and we can use it in tandem with an analog system.

**Comment by colanzio5** [comment]
Yeah I agree about the javascript client, no need to rush that. I have no idea how well command w/ a javascript client would go, so I'll just build out something and let ya'll know when it's in a decent place and we can go from there. 

Still, py2exe isn't an option if we want to use python3, so the issue in this thread still stands - **we need to find another library besides py2exe, if we want to be able to deliver a nice and packed format**

**Comment by alefir** [comment]
I've been able to use [cx_Freeze](https://anthony-tuininga.github.io/cx_Freeze/) on Python3 scripts before.

**Comment by colanzio5** [comment]
@alefir  I saw that one too. Will experiment with other libraries today.

**Comment by waltoncade** [comment]
I have a feeling there are some good python3 exe out there haha
